### DISCUSSION TITLE
draw flow direction

### DISCUSSION THREAD

**Discussion by 4tler** [mapsforge/vtm#864] [Q&A] [answered]
Is it possible to mark the flow direction of rivers? According to OSM, flowing waters should be mapped in flow direction. An extra attribute is not necessary for this. But how can I display this with vtm?

**Comment by devemux86** [comment, marked as answer]
Water lines can have direction (like the road network).
Then you could use something similar to the oneway style.

**Reply by 4tler** [reply]
> Water lines can have direction (like the road network).
> Then you could use something similar to the oneway style.

If I understand correctly, you mean the oneway tag.  
In my opinion, this only regulates legally regulated directions of travel, which are often set in the direction of flow, but do not allow me to define a direction of flow for each watercourse (river, canal, ...).

**Comment by 4tler** [comment]
ok, it looks like the symbols are automatically aligned in the mapping direction. 

thanks :)

example: 

```
<m v="river" zoom-min="7" closed="no">
                <line stroke="#4DCCE8" width="1.5" />
                <m zoom-min="7">
                    <text k="name" style="bold" size="12" fill="#77A5FC" stroke="#E5FFFFFF" stroke-width="2" priority="0"/>
                    <symbol repeat="true" src="file:/symbols/Arrow_Blue_Right_001.svg"/>
                </m>
            </m>
```
![image](https://user-images.githubusercontent.com/74738662/127155191-157526c5-45e0-48e0-9d58-bbd3ffd42edc.png)
